File: Graph/graph.py
#!/usr/bin/env python

# This is a adjacency list representation of a directed graph.

import sys

class Vertex(object):
    def __init__(self, data=None, next_node=None):
        self.data = data
        self.next_node = next_node

class Graph(object):
    def __init__(self, nodes=0):
        self.nodes = nodes
        self.arr = {}

    def add_edge(self, src, dest):
        self.arr[src].append(dest)

    def remove_edge(self, src, dest):
        for node in self.arr.keys():
            if self.arr

    def bfs(self):
        pass

    def dfs(self):
        pass

File: Graph/Graph.java
import java.io.*;
import java.util.*;

/*
 * Implementation of Graph data structure.
 *
 */
public class Graph() {
    
    private int vertex;
    private LinkedList<Integer> adj[];

    /*
     * Constructor.
     *
     */
    Graph(int vertex) {
        this.vertex = vertex;
        this.adj = new LinkedList<Integer>[vertex];
        for(int i=0; i<vertex; i++) {
            this.adj[i] = new LinkedList<Integer>();
        }
    }

    /*
     * Return the number of vertices.
     *
     */
    public int countVertex() {
        return this.vertex;
    }

    /*
     * Return the number of edges.
     *
     */
    public int countEdges() {
        continue;
    }

    /*
     * Add a new edge between two vertices.
     *
     */
    public void addEdge(int src, int dest) {
        this.adj[src].add(dest)
    }

    /*
     * Remove an edge from the graph.
     *
     */
    public removeEdge() {
        continue;
    }

    /*
     * Perform breadth first search.
     *
     */
    public void bfs() {

    }

	public static void main(String[] args) {
        Graph gr = new Graph();
        gr.addEdge(0, 1);
        gr.addEdge(0, 2);
        gr.addEdge(2, 0);
        gr.addEdge(2, 3);
        gr.addEdge(3, 3);
        gr.addEdge(1, 2);
        
        gr.bfs();
	}
}
